File: Easy/Mars Lander - Level 1/C++/Mars Lander - Level 1.cpp
#include <iostream>

using namespace std;

int main()
{
	int numberOfSurfacePoints; // used to draw the surface of Mars
	cin >> numberOfSurfacePoints; cin.ignore();
	for (int i = 0; i < numberOfSurfacePoints; i++) {
		int surfaceCoordinateX; // 0 to 6999
		int surfaceCoordinateY; // 0 to 2999
		cin >> surfaceCoordinateX >> surfaceCoordinateY; cin.ignore();
	}

	while (1) {
		int landerCoordinateX; // in meters
		int landerCoordinateY; // in meters
		int horizontalSpeed; /// in m/s, can be negative
		int verticalSpeed; // in m/s, can be negative
		int fuelRemainig; // in liters
		int rotationAngle; // in degrees (-90 to 90)
		int thrustPower; // 0 to 4
		cin >> landerCoordinateX >> landerCoordinateY >> horizontalSpeed >> verticalSpeed >> fuelRemainig >> rotationAngle >> thrustPower; cin.ignore();

		cout << "0 " << ((abs(verticalSpeed) > 39) ? "4" : "0") << endl;
	}
}

File: Easy/Mars Lander - Level 1/Ruby/MarsLanderLevel1.rb
STDOUT.sync = true

@number_of_surface_points = gets.to_i # used to draw the surface of Mars
@number_of_surface_points.times do
  surface_coordinate_X, surface_coordinate_Y  = gets.split(" ").collect {|x| x.to_i} # 0 to 6999 and  0 to 2999
end

loop do
  # in meters, in meters, in m/s, can be negative, in m/s, can be negative, in liters, in degrees (-90 to 90), 0 to 4
  lander_coordinate_X, lander_coordinate_Y, horizontal_speed, vertical_speed, fuel_remainig, rotation_angle, thrust_power  = gets.split(" ").collect {|x| x.to_i}
  puts "0 " + (vertical_speed.abs > 39 ? "4" : "0")
end